app/popup.tsx

```diff
@@ -11,18 +13,26 @@ const isChromeUrl = (url : string) => {
 
 type ButtonLabel = 'Start Listening' | 'Stop Listening';
 interface PopupProps { enabled: boolean, tabId: number, tabUrl: string };
-interface PopupState { enabled: boolean, label: ButtonLabel, errorMessage?: string };
+interface PopupState { enabled: boolean, label: ButtonLabel, errorMessage?: string, requests: Array<[]> };
 
 class Popup extends React.Component<PopupProps, PopupState> {
   constructor(props: PopupProps) {
     super(props);
     this.state = {
       enabled: props.enabled,
       label: props.enabled ? 'Stop Listening' : 'Start Listening',
-      errorMessage: ''
+      errorMessage: '',
+      requests: [],
     };
   }
 
+  componentDidMount() {
+    MessageService.getRequests(this.props.tabId, (requests) => {
+      console.log('reqs :: ', requests);
```

Comment:
can remove the logs

Reply:
Will do the changes and push accordingly